src/synthesizer/extensions/integration.cpp

```
@@ -29,10 +64,7 @@ static double *trapz_last_axis_serial(double *x, double *y, npy_intp n,
   double *integral = (double *)calloc(num_elements, sizeof(double));
 
   for (npy_intp i = 0; i < num_elements; ++i) {
-    for (npy_intp j = 0; j < n - 1; ++j) {
-      integral[i] +=
-          0.5 * (x[j + 1] - x[j]) * (y[i * n + j + 1] + y[i * n + j]);
-    }
+    integral[i] = trapz_1d(x, y + i * n, static_cast<size_t>(n));
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)